backend/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/fixsuggestions/FixSuggestionsApi.java

```diff
@@ -45,8 +46,12 @@ public AiSuggestionResponseBodyDto getAiSuggestion(AiSuggestionRequestBodyDto dt
       : helper.post("/api/v2/fix-suggestions/ai-suggestions", JSON_CONTENT_TYPE, gson.toJson(dto), cancelMonitor)) {
       return gson.fromJson(response.bodyAsString(), AiSuggestionResponseBodyDto.class);
     } catch (Exception e) {
-      LOG.error("Error while generating an AI CodeFix", e);
-      throw new UnexpectedBodyException(e);
+      if(e instanceof TooManyRequestsException) {
+        throw e;
+      } else {
+        LOG.error("Error while generating an AI CodeFix", e);
+        throw new UnexpectedBodyException(e);
+      }
```

Comment:
Instead of using instanceof, you could catch the 2 different exceptions with 2 catch blocks, I would say it's a more common practice